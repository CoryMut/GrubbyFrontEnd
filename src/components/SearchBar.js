import React from "react";
import { makeStyles } from "@material-ui/core/styles";
import SearchIcon from "@material-ui/icons/Search";
import InputBase from "@material-ui/core/InputBase";
import IconButton from "@material-ui/core/IconButton";
import InputAdornment from "@material-ui/core/InputAdornment";
import CloseIcon from "@material-ui/icons/Close";

const useStyles = makeStyles((theme) => ({
    search: {
        backgroundColor: "white",
        border: "1px solid black",
        borderRadius: "25px",
        textAlign: "center",
    },
    inputInput: {
        padding: theme.spacing(2, 1, 2, 0),
        paddingLeft: `calc(1em + ${theme.spacing(4)}px)`,
        transition: theme.transitions.create("width"),
        width: "100%",
    },
    searchIcon: {
        color: "#b19cd9",
    },
}));

// { handleChange, handleSubmit, searchTerm, submitStatus, handleClear, disabled }

const SearchBar = ({ handleChange, handleClear, handleSubmit, submitStatus, searchTerm, query }) => {
    const classes = useStyles();

    return (
        <div className={classes.search}>
            <form onSubmit={handleSubmit}>
                <InputBase
                    value={searchTerm}
                    onChange={handleChange}
                    placeholder="Search by keyword or character!"
                    classes={{
                        root: classes.inputRoot,
                        input: classes.inputInput,
                    }}
                    fullWidth
                    inputProps={{
                        "aria-label": "search",
                    }}
                    endAdornment={
                        <InputAdornment position="start">
                            {/* <IconButton onClick={submitStatus ? handleClear : handleSubmit}> */}
                            <IconButton onClick={query ? handleClear : handleSubmit}>
                                {query ? (
                                    <CloseIcon className={classes.searchIcon} />
                                ) : (
                                    <SearchIcon className={classes.searchIcon} />
                                )}
                            </IconButton>
                        </InputAdornment>
                    }
                />
            </form>
        </div>
    );
};

export default SearchBar;
