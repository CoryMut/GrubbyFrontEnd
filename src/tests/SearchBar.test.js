import React from "react";
import { shallow } from "enzyme";
import SearchBar from "../components/SearchBar";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router";

jest.mock("react-router-dom", () => ({
    useHistory: () => ({
        push: jest.fn(),
    }),
    useLocation: () => ({
        push: jest.fn(),
    }),
}));

test("should test NavBar component", () => {
    const wrapper = shallow(<SearchBar />);
    expect(wrapper).toMatchSnapshot();
});

test("it renders without crashing", () => {
    render(
        <MemoryRouter>
            <SearchBar />
        </MemoryRouter>
    );
});
