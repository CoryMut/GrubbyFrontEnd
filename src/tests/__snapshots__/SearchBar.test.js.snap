// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test NavBar component 1`] = `
<div
  className="makeStyles-search-1"
>
  <form>
    <WithStyles(ForwardRef(InputBase))
      classes={
        Object {
          "input": "makeStyles-inputInput-2",
          "root": undefined,
        }
      }
      endAdornment={
        <ForwardRef(WithStyles)
          position="start"
        >
          <ForwardRef(WithStyles)>
            <UNDEFINED
              className="makeStyles-searchIcon-3"
            />
          </ForwardRef(WithStyles)>
        </ForwardRef(WithStyles)>
      }
      fullWidth={true}
      inputProps={
        Object {
          "aria-label": "search",
        }
      }
      placeholder="Search by keyword or character!"
    />
  </form>
</div>
`;
